### ISSUE TITLE
Issues with References in Empty Assumptions Section

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#1415]
While adding an empty assumptions section to Projectile, I noticed that this was the intro for the section:

![assumps](https://user-images.githubusercontent.com/35857611/58428522-c5848680-8070-11e9-924d-686566d6ee5e.png)

It references a lot of other sections that don't exist, as per #946, but it also talks about the "numbers in square brackets", which is only true in the LaTeX, not the HTML. What should be done about this?

**Comment by Mornix** [comment]
Overlaps with #1164.

**Comment by JacquesCarette** [comment]
This basically means that the way that the text, and in fact the whole introduction section, is constructed is too hard-coded. That was indeed already spotted before, and this is worse. This whole part needs to be generated more dynamically.

**Comment by smiths** [comment]
Yes, this is a good issue.  Fixing the parameterized boiler-plate and "roadmap" text is an example that @szymczdm should be able highlight in his paper.

**Comment by cd155** [comment]
There is a ticket talk about determining the actual document outline base on content. This is an example of #946.
